### PR TITLE
[Skia] Do not use the legacy native GL/EGL interface factory

### PR DESCRIPTION
**Pull request by aperezdc**
#### c6ffba6807321620a68303f1d28aab36b5e012a1
<pre>
[Skia] Do not use the legacy native GL/EGL interface factory
<a href="https://bugs.webkit.org/show_bug.cgi?id=270047">https://bugs.webkit.org/show_bug.cgi?id=270047</a>

Reviewed by Carlos Garcia Campos.

Explicitly instantiate a GrGLInterface used for creating the Skia GL
context. This is done in a helper function which may return the Epoxy or
the plain EGL interface depending on USE(LIBEPOXY). In any case, when
USE(EGL) is not effect for now this produces a build error for now.
Ports that want to use the EGL instead of the Epoxy factory will still
need some CMake work to build Skia&apos;s EGL native interface instead of the
Epoxy one in a follow-up patch, this only leaves the instantiation code
ready for that.

* Source/ThirdParty/skia/CMakeLists.txt: Do not build GrGLMakeNativeInterface_epoxy.cpp
* Source/ThirdParty/skia/src/gpu/ganesh/gl/epoxy/GrGLMakeNativeInterface_epoxy.cpp: Removed.
* Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp:
(WebCore::skiaGLInterface): Added. Helper function that returns the correct
GrGLInterface for the current build configuration.
(WebCore::PlatformDisplay::skiaGLContext): Pass the GrGLInterface
returned by skiaGLInterface() when instantiating the Skia GL context.

Canonical link: <a href="https://commits.webkit.org/275311@main">https://commits.webkit.org/275311@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/d7a5bb8b9850e941976c34c5d133e2416015d363

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41496 "Failed to checkout and rebase branch from PR 25069") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20510 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43874 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44063 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37588 "Built successfully") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/43803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23616 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17840 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/34310 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42070 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/17438 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/35740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/14970 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/15154 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/36743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45423 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/37685 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/37063 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/40815 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16311 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/13389 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/39223 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/17930 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17985 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5544 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/17574 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | |

### CODE REVIEW

**Comment by aperezdc** [comment]
This came up in the review for my submission to upstream Skia of the [Epoxy native interface patch](https://skia-review.googlesource.com/c/skia/+/814176/), where `GrGLMakeNativeInterface_epoxy.cpp` also got removed.

When we update Skia to a version that includes the Epoxy native interface we will need to change the `GrGLMakeEpoxyEGLInterface()` call to use `GrGLInterfaces::MakeEpoxyEGL()`&mdash;the name change is for consistency with the rest of renames being applied to Skia.

File: Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp [hunk 1]
```diff
@@ -39,8 +39,35 @@ IGNORE_CLANG_WARNINGS_END
...
 
+#if USE(EGL)
+#if USE(LIBEPOXY)
+#include <skia/gpu/gl/epoxy/GrGLMakeEpoxyEGLInterface.h>
+#else
+#include <skia/gpu/gl/egl/GrGLMakeEGLInterface.h>
+#endif
+#endif
+
+#include <wtf/NeverDestroyed.h>
```
Comment (carlosgcampos) [inline Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp]: Move this to the main includes block

File: Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp [hunk 2]
```diff
@@ -39,8 +39,35 @@ IGNORE_CLANG_WARNINGS_END
...
+
 namespace WebCore {
 
+static sk_sp<const GrGLInterface> getSkiaGLInterface()
```
Comment (carlosgcampos) [inline Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp]:
```suggestion
static sk_sp<const GrGLInterface> skiaGLInterface()
```

**Review by carlosgcampos — APPROVED** [review]

File: Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp (hunk 1 above)
Comment (aperezdc) [reply Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp]: Done.

File: Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp [hunk 3]
```diff
@@ -39,8 +39,35 @@ IGNORE_CLANG_WARNINGS_END
 #include <skia/gpu/gl/GrGLInterface.h>
 #include <skia/gpu/gl/GrGLTypes.h>
```
Comment (carlosgcampos) [inline Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp]: Why this empty line? We normally use a single block of includes

**Review by carlosgcampos — APPROVED** [review]

File: Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp (hunk 3 above)
Comment (aperezdc) [reply Source/WebCore/platform/graphics/skia/PlatformDisplaySkia.cpp]: No reason, it's just the style checker didn't catch it. I will remove it before landing.

**Pull request merged**